src/core/search/query_driver.h

```diff
@@ -20,17 +20,14 @@ class QueryDriver {
   QueryDriver();
   ~QueryDriver();
 
-  Scanner* scanner() {
-    return scanner_.get();
-  }
-
   void SetInput(std::string str) {
     cur_str_ = std::move(str);
     scanner()->in(cur_str_);
   }
 
-  void SetParams(QueryParams params) {
-    params_ = std::move(params);
+  void SetParams(const QueryParams& params) {
+    params_ = &params;
```

Comment:
If you're going to save `params` for longer than the function call, maybe take it as a pointer?

Reply:
I see that you do it for Scanner below, any reason not to do it here as well?

Reply:
True